### PR TITLE
IGNITE-18265 Update Raft packages structure

### PR DESCRIPTION
**Pull request by sashapolo**
* Renamed `ignite-raft-client` to `ignite-raft-api`
* Got rid of JRaft classes leaking from `ignite-raft` as much as possible

https://issues.apache.org/jira/browse/IGNITE-18265

### CODE REVIEW

**Review by kgusakov — CHANGES_REQUESTED** [review]

File: modules/affinity/pom.xml [hunk 1]
```diff
@@ -33,39 +33,17 @@
     <version>3.0.0-SNAPSHOT</version>
 
     <dependencies>
-        <dependency>
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/affinity/pom.xml:36]: yes

File: modules/metastorage-client/src/integrationTest/java/org/apache/ignite/internal/metastorage/client/ItMetaStorageServiceTest.java [hunk 2]
```diff
@@ -908,38 +895,32 @@ public void testCursorsCleanup() throws Exception {
...
 
-            assertThrows(NoSuchElementException.class, () -> cursorNode0.iterator().next());
+        assertTrue(cursorNode1.hasNext());
 
-            assertThrows(NoSuchElementException.class, () -> cursor2Node0.iterator().next());
+        metaStorageSvc.closeCursors(NODE_ID_0).get();
 
-            assertEquals(EXPECTED_RESULT_ENTRY, (cursorNode1.iterator().next()));
-        } finally {
-            metaStorageRaftGrpSvc.shutdown();
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/metastorage-client/src/integrationTest/java/org/apache/ignite/internal/metastorage/client/ItMetaStorageServiceTest.java:941]: my bad, let's bring it back

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java [hunk 3]
```diff
@@ -769,7 +775,7 @@ private void updateAssignmentInternal(ConfigurationNotificationEvent<byte[]> ass
                                         );
 
                                         try {
-                                            raftMgr.startRaftGroupNode(
+                                            ((Loza) raftMgr).startRaftGroupNode(
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:778]: yes, I will add it to the ticket

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/exec/MockedStructuresTest.java [hunk 4]
```diff
@@ -511,7 +508,7 @@ private TableManager createTableManager() {
                 workDir,
                 msm,
                 schemaManager,
-                view -> new LocalLogStorageFactory(),
+                null,
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/exec/MockedStructuresTest.java:511]: This callback is only invoked if an in-memory Raft node has been configured

File: modules/runner/build.gradle [hunk 5]
```diff
@@ -32,6 +32,7 @@ dependencies {
     implementation project(':ignite-api')
     implementation project(':ignite-vault')
     implementation project(':ignite-network')
+    implementation project(':ignite-raft-api')
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/runner/build.gradle:35]: maven build works correctly without this dependency as it is transitive

File: modules/table/build.gradle [hunk 6]
```diff
@@ -33,7 +33,7 @@ dependencies {
     implementation project(':ignite-network-api')
     implementation project(':ignite-raft')
     implementation project(':ignite-replicator')
-    implementation project(':ignite-raft-client')
+    implementation project(':ignite-raft-api')
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/table/build.gradle:36]: please see my comment above

File: modules/runner/build.gradle (hunk 5 above)
[the remark this answers is not in the document]
Comment (kgusakov) [reply modules/runner/build.gradle:35]: Ok, thanks

File: modules/metastorage-client/src/integrationTest/java/org/apache/ignite/internal/metastorage/client/ItMetaStorageServiceTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/metastorage-client/src/integrationTest/java/org/apache/ignite/internal/metastorage/client/ItMetaStorageServiceTest.java:941]: fixed

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java:779]: fixed

**Review by kgusakov — APPROVED** [review]

**Pull request merged**